safe_control_gym/envs/gym_control/cartpole.py

```diff
@@ -584,8 +584,15 @@ def _get_observation(self):
         # Wrap angle to constrain state space, useful in swing-up task.
         if self.obs_wrap_angle:
             obs[2] = normalize_angle(obs[2])
+
         # Concatenate goal info (goal state(s)) for RL
-        obs = self.extend_obs(obs, self.ctrl_step_counter+1)
+        # Plus two because ctrl_step_counter has not incremented yet, and we want to return the obs (which would be
+        # ctrl_step_counter + 1 as the action has already been applited), and the next state (+ 2) for the RL to see
+        # the next state.
+        if self.at_reset:
+            obs = self.extend_obs(obs, self.ctrl_step_counter+1)
```

Comment:
Make this ```obs = self.extend_obs(obs, 1)```

Reply:
@Justin-Yuan are we correctly interpreting that the `extend_obs` method takes `self.ctrl_step_counter+2` as second argument except on reset when it takes always `1` (i.e. indexes the second element in the reference)?